src/apps/amp/ethereum/amp.farm.contract-position-fetcher.ts

```diff
@@ -63,30 +81,42 @@ export class EthereumAmpFarmContractPositionFetcher extends SingleStakingFarmTem
       });
   }
 
-  getContract(address: string): AmpStaking {
-    return this.contractFactory.ampStaking({ address, network: this.network });
-  }
-
-  async getFarmDefinitions(): Promise<SingleStakingFarmDefinition[]> {
-    return FARMS;
-  }
-
-  async getRewardRates() {
-    return [0];
-  }
-
-  async getStakedTokenBalance({ address }: GetTokenBalancesParams<AmpStaking>) {
-    // We rely on the API to get staked and claimable balances
-    const { supplyTotal, rewardTotal } = await this.getApiData(address);
-    return new BigNumber(supplyTotal).plus(rewardTotal).toFixed(0);
+  async getTokenBalancesPerPosition({ address }: GetTokenBalancesParams<AmpStaking>): Promise<BigNumberish[]> {
+    const { supplyTotal, rewardTotal } = await this.getAddressBalances(address);
```

Comment:
Is there a way to get balances on chain ?

Reply:
There isn't, which is why we use our API.